sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_floating_point.asciidoc

```diff
@@ -0,0 +1,1280 @@
+= SPV_INTEL_arbitrary_precision_floating_point
+
+== Name Strings
+
+SPV_INTEL_arbitrary_precision_floating_point
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+== Contributors
+
+Ajaykumar Kannan, Intel
+
+== Notice
+
+Copyright (c) 2020 Intel Corporation.  All rights reserved.
+
+== Status
+
+Initial Release
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision           | 1
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification Version 1.5, Revision 2.
+
+This extension depends on the SPV_INTEL_arbitrary_precision_int spec.
+
+== Overview
+
+_ihc_float_ is a datatype represented in SPIR-V using `OpTypeInt` that implements arbitrary precision floating point support.
+This datatype and its corresponding operations can be useful on targets that can take advantage of narrower representation such as FPGAs.
+
+The datatype is additionally described with two parameters:
+
+* _E_: represents the number of exponent bits
+* _M_: represents the number of mantissa bits
+
+The total width of the type is `E+M+1` where the last bit is used to represent the sign.
+The data layout is shown below:
+
+`[ S (sign bit) ][ E (Exponent) ][ M (Mantissa) ]`
+
+The width of the data `(E+M+1)` is encoded with the width of the `OpTypeInt`.
```

Comment:
What about `Signedness` operand of `OpTypeInt`? Is it used? Please note that if `Kernel` capability is used in a module, than `Signedness` of `OpTypeInt` must always be 0.

Reply:
> Would you like me to include that snippet in here well? 

I think it worth doing so